wz_strategic_plugin/src/wz_strategic_plugin.cpp

```diff
@@ -0,0 +1,375 @@
+                lanelet::Id end_lane_id = traffic_light_lanelet.second.id();
+
+                auto traffic_light_current_state = traffic_light_interpreter(nearest_traffic_light->predictState(start_time));
+                ROS_DEBUG("traffic_light_current_state %d", traffic_light_current_state);
+
+                if(!traffic_light_current_state) {
+                    return true;
+                }
+
+                auto traffic_light_next_predicted_state = traffic_light_interpreter(nearest_traffic_light->predictState(start_time + time_remaining_to_traffic_light));
+                ROS_DEBUG("traffic_light_next_predicted_state %d", traffic_light_next_predicted_state);
+
+                if(!traffic_light_next_predicted_state) {
+                    return true;
+                }
+
+                if(traffic_light_current_state != traffic_light_next_predicted_state) {
+
+                    if(traffic_light_next_predicted_state == 0) {
+
+                        double estimated_distance_to_stop = estimate_distance_to_stop(current_speed_,declaration);
+                        double estimated_time_to_stop = estimate_time_to_stop(estimated_distance_to_stop,current_speed_);
+
+                        double lane_following_distance = traffic_light_down_track - estimated_distance_to_stop;
+
+                        // fix time
+                        // stop_and_wait
+                        cav_msgs::Maneuver stop_and_wait = composeStopAndWaitManeuverMessage(current_progress + lane_following_distance, 
+                                                traffic_light_down_track, 
+                                                current_speed_, 
+                                                start_lane_id,
+                                                end_lane_id,
+                                                start_time,
+                                                estimated_time_to_stop);
+
+                        resp.new_plan.maneuvers.push_back(stop_and_wait);
+
+                        // workzone_tactical
+                        cav_msgs::Maneuver workzone_tactical = composeIntersectionTransitMessage( traffic_light_down_track,
+                                                                                                  workzone_end_point_down_track, 
+                                                                                                  current_speed_, 
+                                                                                                  current_speed_,
+                                                                                                  start_time + ros::Duration(std::max(15.0,estimated_time_to_stop)),
+                                                                                                  start_lane_id);
+                        resp.new_plan.maneuvers.push_back(workzone_tactical);
+
+                    }
+
+                    if(traffic_light_next_predicted_state == 1) {
```

Comment:
Can we define this 0 and 1 as constant class members that specify what this state means? Just to make it easier to read.
For example if 0 is supposed to show red light state, lets create a const int member in the header RED_LIGHT_STATE_ = 0.
We should also change this second if condition to an "else if" since it is part of the same loop.
There should be an else condition following that which results in a "not valid state" warning.